spaceros/Earthfile

```diff
@@ -92,9 +91,40 @@ setup:
   RUN sudo add-apt-repository ppa:kisak/kisak-mesa
   RUN sudo apt update && sudo apt upgrade -y
 
+spaceros-artifacts:
+  # we must run it in a separate container, so that downstream tasks can be cached if vcs file does not change
+  FROM ubuntu:jammy
+  RUN apt-get update && apt-get install -y git wget
+
+  # main purpose of this command is to make sure that the git ls-remote results are not cached
+  RUN --no-cache echo "Cloning spaceros repo artifacts"
+
+  # current git branch, prefilled by earthly: https://docs.earthly.dev/docs/earthfile/builtin-args
+  ARG EARTHLY_GIT_BRANCH
+
+  ARG SPACEROS_REPO_URL="https://github.com/space-ros/space-ros.git"
+  # if current local branch does not exist in target repo then use main. note that branch supplied from CLI overrides this behavior.
+  ARG SPACEROS_GIT_REF="$( [ -n \"$(git ls-remote $SPACEROS_REPO_URL $EARTHLY_GIT_BRANCH)\" ] && echo $EARTHLY_GIT_BRANCH || echo 'main' )"
```

Comment:
As we want to re-run ls-remote every time we can no longer keep them at the top of the file.